base/linalg/cholesky.jl

```diff
@@ -78,83 +104,149 @@ function chol!{T}(A::AbstractMatrix{T}, ::Type{LowerTriangular})
     return LowerTriangular(A)
 end
 
-"""
-    chol(A::AbstractMatrix) -> U
-
-Compute the Cholesky factorization of a positive definite matrix `A`
-and return the UpperTriangular matrix `U` such that `A = U'U`.
-"""
-function chol{T}(A::AbstractMatrix{T})
```

Comment:
I'd restricted it to `StridedMatrix` to avoid overlap with `Hermitian` but it's not necessary in the version of this I ended up with so I've changed it back to `AbstractMatrix`.